unit_testing/unit_test.py

```diff
@@ -662,3 +665,56 @@
     print(str(sec) + chr(subsec) + " X - Altimetry quick plot not saved")
 
 plt.close('all')
+
+#################################################
+## ( 8 ) TIDEGAUGE Methods                     ##
+#################################################
+sec = sec+1
+subsec = 96
+
+#-----------------------------------------------------------------------------#
+# ( 8a ) Load in GESLA tide gauge files from directory                        #
+#                                                                             #
+subsec = subsec+1
+
+try:
+    date0 = datetime.datetime(2010,1,1)
+    date1 = datetime.datetime(2010,12,1)
+    tg = coast.TIDEGAUGE(dn_tidegauge, date0, date1)
+    
+    # Check length of dataset_list is correct and that
+    if len(tg.dataset_list) == 9:
+        print(str(sec) + chr(subsec) + " OK - Tide gauges loaded")
+except:
+    print(str(sec) + chr(subsec) +' FAILED.')
+
```

Comment:
It is not obvious how to access the tide gauge data. To make this clear add in a simple test in addition to the len() == 9, perhaps combine the print statement since they are trivial tests:

try:
if tg.dataset_list[0].attrs == {'site_name': 'FUKAURA',
 'country': 'Japan',
 'contributor': 'Japan_Meteorological_Agency',
 'latitude': 40.65,
 'longitude': 139.9333,
 'coordinate_system': 'Unspecified',
 'original_start_date': Timestamp('1971-12-31 15:00:00'),
 'original_end_date': Timestamp('2013-12-31 14:00:00'),
 'time_zone_hours': 0.0,
 'precision': 0.01,
 'null_value': -99.9999}

      print(str(sec) + chr(subsec) + " OK - Tide gauges loaded and sample of attributes checked")

Reply:
Done